contracts/Tokens/Prime/Prime.sol

```diff
@@ -265,8 +268,18 @@ contract Prime is IIncomeDestination, AccessControlledV8, PausableUpgradeable, M
 
         if (isIrrevocable) {
             for (uint i = 0; i < users.length; ) {
-                _mint(true, users[i]);
-                _initializeMarkets(users[i]);
+                Token storage userToken = tokens[users[i]];
+                if (userToken.exists && !userToken.isIrrevocable) {
+                    //upgrade to irrevocable token
+                    userToken.isIrrevocable = true;      
+                    totalIrrevocable++;
+                    totalRevocable--; 
```

Comment:
I think we should check if we are passing the threshold `totalIrrevocable` after doing the upgrade. I think I would extract this logic to a new internal function, for the sake of readability